app/exerlog/lib/src/widgets/gradient_button.dart

```diff
@@ -62,4 +56,14 @@ class _RaisedGradientButtonState extends State<RaisedGradientButton> {
       ),
     );
   }
+
+  void Function()? _onPressed() {

```

Comment:
I would refactor this method to something like this, makes it a little clearer in my opinion:

```
  Future<void> _onPressed() async {
    if (_isLoading) return;
    setState(() => _isLoading = true);
    await widget.onPressed();
    setState(() => _isLoading = false);
  }
```